stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,367 @@
+    def onchange_warehouse_id(self):
+        self.from_picking_type_id = self.warehouse_id.out_type_id
+
+    def _generate_delivery_route(self):
+        self.ensure_one()
+        if self.delivery_route_id:
+            return False
+        # Re-generate the WH configuration with the chosen delivery steps in a
+        # savepoint to get a whole route+rules+picking types configured almost
+        # automatically (to not re-invent the wheel).
+        # Delivery route values are then copied to generate a new one.
+        # Copy some picking types with new sublocations first so the new
+        # delivery route will use them.
+        if not self.out_type_id:
+            self.out_type_id = self.warehouse_id.out_type_id.copy(
+                {
+                    "name": (
+                        f"{self.warehouse_id.out_type_id.name} "
+                        f"{self.sequence_prefix}"
+                    ),
+                    "active": True,
+                }
+            )
+            self.out_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+        if "pick_" in self.delivery_steps:
+            if not self.pick_type_id:
+                self.pick_type_id = self.warehouse_id.pick_type_id.copy(
+                    {
+                        "name": (
+                            f"{self.warehouse_id.pick_type_id.name} "
+                            f"{self.sequence_prefix}"
+                        ),
+                        "active": True,
+                    }
+                )
+                self.pick_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+            # Create a dedicated 'Output/SUB' location
+            if not self.output_stock_loc_id:
+                self.output_stock_loc_id = (
+                    self.warehouse_id.wh_output_stock_loc_id.copy(
+                        {
+                            "name": self.sequence_prefix,
+                            "location_id": self.warehouse_id.wh_output_stock_loc_id.id,
+                        }
+                    )
+                )
+                self.pick_type_id.default_location_dest_id = self.output_stock_loc_id
+                self.out_type_id.default_location_src_id = self.output_stock_loc_id
+            if "pack_" in self.delivery_steps:
+                if not self.pack_type_id:
+                    self.pack_type_id = self.warehouse_id.pack_type_id.copy(
+                        {
+                            "name": (
+                                f"{self.warehouse_id.pack_type_id.name} "
+                                f"{self.sequence_prefix}"
+                            ),
+                            "active": True,
+                        }
+                    )
+                    self.pack_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+                # Create a dedicated 'Packing Zone/SUB' location
+                self.pack_stock_loc_id = self.warehouse_id.wh_pack_stock_loc_id.copy(
```

Comment:
In 3 steps delivery, when you set a pack_stock_loc_id but not a pick type, the pack_stock_loc_id location is overridden by a newly generated one.
This should not occur.
So there is something weird being the scene after doing https://github.com/OCA/wms/pull/431/files#diff-ec16197c7c77f54e4580085fc7fa4bfe23625085a978e8658cd0835e253423a1R270